nixos/modules/programs/wayland/hyprland.nix

```diff
@@ -25,18 +24,26 @@ in
       '';
     };
 
-    package = mkPackageOptionMD pkgs "hyprland" { };
+    package = mkPackageOptionMD pkgs "hyprland" {
+      nullable = true;
```

Comment:
Good catch.

I made the package option nullable here to match the Sway and River modules.
The goal was to allow enabling both the nixpkgs module to set system-wide options, and the HM module to install the compositor itself, "so different users can have different versions / package overrides" (see [here](https://github.com/NixOS/nixpkgs/pull/213482#issuecomment-1647907857)).

Reply:
However for Hyprland the portal package depends on the compositor package, so we should logically add one portal package per compositor package installed… which doesn't sound very good.
It's simpler to avoid supporting this edge case, and as you said just assume the users will use the same options everywhere, so I guess I'll revert this change.